source/MaterialXShaderGen/HwShaderGenerator.cpp

```diff
@@ -8,7 +8,7 @@ namespace MaterialX
 
 HwShaderGenerator::HwShaderGenerator(SyntaxPtr syntax)
     : ShaderGenerator(syntax)
-    , _maxActiveLightSources(3)
+    , _maxActiveLightSources(0)
```

Comment:
The max test done for light handler takes the max of this and the # of lights which could end up with more lights than actually exists. So setting default to 0.

Reply:
This is the _max_ active lights supported by generated shaders. So not the same as the actual number of lights used which is controlled by the uniform u_numActiveLightSources. Setting max to 0 by default means a client _must_ call HwShaderGenerator::setMaxActiveLightSources(N) in order to get any lighting. So I think I would prefer having the max lights to something higher than zero by default. I'm not sure if 3 is the best possible default though :)

Reply:
Let me double check. I think the issue is that the max 3 lights are created but if there are less than 3 (say 1) then the other 2 have undefined values.

Reply:
Yes the u_lightData[] array will hold _max_ number of lights, so the unused entries will hold undefined values, but they should never be accessed by the shader. The uniform u_numActiveLightSources should be set to the actual number of used lights to make sure that.

The idea is that the same shader should be usable with different sets of light sources without a recompile (up the the _max_ limit).

Reply:
Maybe u_numActiveLightSources is not set when binding the lights?